### PR TITLE
Chore: Remove debounce on analyze

### PR DESCRIPTION
**Pull request by idillon-sfl**
Tree-sitter claims to be [`Fast enough to parse on every keystroke in a text editor`](https://tree-sitter.github.io/tree-sitter/).

By removing the debouncing on the analysis, we get a better responsiveness. It actually improves (fixes?) the completion in embedded languages.

### CODE REVIEW

**Review by deribaucourt — APPROVED** [review]
CPU usage is contained, no behavioral regression. Nice!

**Comment by idillon-sfl** [comment]
> CPU usage is contained, no behavioral regression. Nice!

Lets just wait for @WilsonZiweiWang approval, he's the most familiar with that part of the code.

**Review by WilsonZiweiWang — APPROVED** [review]

**Pull request merged**
